### ISSUE TITLE
Threads with unread posts sometimes open the wrong page (site bug)

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#642]
Some threads have magical missing posts, which affects the ``replies`` count that we scrape to calculate the postcount for a thread (``replies + 1`` for the first post). When you've read a full page, clicking the thread should take you to the next unread post, i.e. the first one on the next page.

But because the postcount is 1 lower than it should be, the index of the next unread post is 1 lower too - so it's actually calculated as the last post on the page you already read. So it will always load that page, until you manually turn the page and view more posts (and if you view another full page, the problem will appear again)

-------------------

I'm mostly adding this for information, in case anyone else runs into it like I did. Right now there's no other info we can scrape, so we have to rely on the (potentially incorrect) reply numbers.

I guess it would be _possible_ to calculate a postcount from the actual thread view, when you're looking at the last page (``(previousPages * postsPerPage) + postsOnThisPage``) - which we already do - and compare that to the count on the thread list, updating a correction offset column in the thread database. But that's pretty complicated, especially ensuring no new posts arrived between loading one page and the other

**Comment by Sereri** [comment]
The cause of this in most cases is Zen Death Robot's posts getting deleted by the post fairy

**Comment by baka-kaba** [comment]
Yeah but I just saw it happening in the very new Trump thread, hasn't ZDR been missing a while?
